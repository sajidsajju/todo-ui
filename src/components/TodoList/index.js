import React, { useState } from "react";
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles({
  todo: {
    width: "40em",
    margin: "auto",
    background: "#FFFFFF",
  },
  wrapper: {
    paddingLeft: "10px",
    fontSize: "1.5em",
    lineHeight: "2em",
    display: "inline",
  },
  input: {
    border: "none ",
    outline: "none",
    fontSize: "1em",
    width: "72%",
    "&:active, &:focus": {
      color: "#00BFFF",
    },
    DoneOutlineIcon: {
      fontSize: "55em",
    },
  },
});

const List = props => {
  const classes = useStyles();
  const { todo } = props;

  const [text, setText] = useState("");
  const [inputField, setInputField] = useState(false);

  return (
    <div className={classes.todo}>
      <div className={classes.wrapper}>
        {inputField ? (
          <input
            type="text"
            className={classes.input}
            value={text}
            onChange={e => setText(e.target.value)}
            data-testid="input-field"
          />
        ) : (
          todo.text
        )}
      </div>
      <hr />
    </div>
  );
};

function TodoList(props) {
  const { todos } = props;

  return (
    <>
      {todos.map((todo, index) => (
        <List todo={todo} key={index} />
      ))}
    </>
  );
}

export default TodoList;
