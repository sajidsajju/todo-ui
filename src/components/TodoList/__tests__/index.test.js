import React from "react";
import { fireEvent, render } from "@testing-library/react";

import TodoList from "../";

describe("<TodoList  />", () => {
  const todos = [{ text: "hello" }, { text: "great" }];

  it("should match the snapshot", () => {
    const rendered = render(<TodoList todos={todos} />);

    expect(rendered.container.firstChild).toMatchSnapshot();
  });
});
