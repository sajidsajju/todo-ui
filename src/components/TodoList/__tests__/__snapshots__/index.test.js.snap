// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList  /> should match the snapshot 1`] = `
<div
  class="makeStyles-todo-1"
>
  <div
    class="makeStyles-wrapper-2"
  >
    hello
  </div>
  <hr />
</div>
`;
